tasks/gpsr/find_object/src/object_comparison.py

```diff
@@ -0,0 +1,175 @@
+
+        def count_types(self, detections):
+            """
+            Count the number of different object types in detections.
+
+            :param detections: List of detection tuples (object_type, position)
+            :return: Dictionary with object types as keys and counts as values
+            """
+            # userdata.detections_3d = [(detection, point),(detection, point),(detection, point)]
+            print("we're counting")
+            object_counts = {}
+            for detection in detections:
+                object_type = detection.name
+            
+                if len(object_counts.keys()) == 0:
+                    object_counts[object_type] = 1
+                else:
+                    if object_type in object_counts.keys():
+                        object_counts[object_type] += 1
+                    else:
+                        print("yay")
+                        object_counts[object_type] = 1
+            return object_counts
+
+        def execute(self, userdata):
+            print("we're doing cout types")
+            print(userdata.detections_3d)
+            filtered_detections = userdata.detections_3d # the outcome of the 3d detections
+            rospy.loginfo(filtered_detections)
+            rospy.loginfo(type(filtered_detections.detected_objects[0]))
+            object_counts = self.count_types(filtered_detections.detected_objects)
+            print("finished counting")
+            print(object_counts)
+            userdata.object_dict = object_counts # 1.Tell me how many of an object (or object category) are on a placement location.
+            userdata.detections_types = object_counts.keys() # 2.Find a (one or three) category of object in a room.
+            return 'succeeded'
+    
+    class ObjectWeight(smach.State):
+        def __init__(self):
+            smach.State.__init__(self, outcomes=['succeeded', 'failed'], input_keys=['detections_types'], output_keys=['sorted_weights'])
+
+        def get_weight(self, detections):
+            """
+            rank the weight of a list of objects.
+
+            :param detections: List of detection tuples (object_type, position)
+            :return: a list of the weight rank of object in a category(max->min)
+            """
+            average_weights = {
+                    "sponge": 10,
+                    "cleanser": 850,
+                    "tennis ball": 58,
+                    "soccer ball": 430,
+                    "dice": 9,
+                    "baseball":145,
+                    "rubiks_cube": 100,
+                    "apple": 182,
+                    "orange": 131,
+                    "pear":178,
+                    "banana":118,
+                    "strawberry":12,
+                    "lemon":100,
+                    "peach":150,
+                    "plum":78,
+                    "bottle":500,
+                    "cup":300
+                }
```

Comment:
I feel like this dictionary should be passed in the constructor of `ObjectComparison` (which then should pass it to the constructor of `ObjectWeight`